mettle/src/stdapi/net/client.c

```diff
@@ -349,6 +349,10 @@ udp_client_read(struct channel *c, void *buf, size_t len)
 		return -1;
 	}
 	void *msg_buf = network_client_read_msg(ucc->nc, &msg_len);
+	if (msg_buf == NULL) {
+		errno = EIO;
+		return -1;
+	}
```

Comment:
This was the core bug right here. Then we return `-1` and check it appropriately in `channel_set_interactive`.